flink-table-store-common/src/main/java/org/apache/flink/table/store/file/predicate/PredicateConverter.java

```diff
@@ -91,15 +91,17 @@ public Predicate visit(CallExpression call) {
             for (int i = 1; i < children.size(); i++) {
                 literals.add(extractLiteral(fieldRefExpr.getOutputDataType(), children.get(i)));
             }
-            return builder.in(fieldRefExpr.getInputIndex(), literals);
+            return builder.in(builder.indexOf(fieldRefExpr.getName()), literals);
```

Comment:
It is safer to use name. The previous implementation has problems, because if the project pushdown is before and the filter pushdown is after, the index of the filter will be projected.